### DISCUSSION TITLE
is:open umi3.5 项目越来越复杂，业务组件越来越多，比较大了，文件过多，开发第一次打包启动很慢 ，足足有20分钟。这个要什么优化呢。

### DISCUSSION THREAD

**Discussion by tianyazjq** [umijs/umi#10993] [General]
is:open umi3.5 项目越来越复杂，业务组件越来越多，比较大了，文件过多，开发第一次打包启动很慢 ，足足有20分钟。这个要什么优化呢。
类似实现vite的那种1分钟内的快速启动。路由的动态加载编译。

**Comment by xierenyuan** [comment]
要不要考虑升级到 umi4 看看 (umi4 同时还支持 vite)

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4

**Comment by coder-belong** [comment]
可以升级到 umi4，umi4 集成了 mako，不过可惜的是 mako 不支持 window环境
